areas/appservice/tests/AzureMcp.AppService.UnitTests/Commands/Database/DatabaseAddCommandTests.cs

```diff
@@ -0,0 +1,362 @@
+        var databaseName = "test-db";
+        var tenant = "tenant123";
+
+        var expectedConnection = new DatabaseConnectionInfo
+        {
+            DatabaseType = databaseType,
+            DatabaseServer = databaseServer,
+            DatabaseName = databaseName,
+            ConnectionString = "Server=test-server.database.windows.net;Database=test-db;Trusted_Connection=True;TrustServerCertificate=True;",
+            ConnectionStringName = "test-dbConnection",
+            IsConfigured = true,
+            ConfiguredAt = DateTime.UtcNow
+        };
+
+        _appServiceService.AddDatabaseAsync(
+            Arg.Is(appName),
+            Arg.Is(resourceGroup),
+            Arg.Is(databaseType),
+            Arg.Is(databaseServer),
+            Arg.Is(databaseName),
+            Arg.Any<string>(),
+            Arg.Is(subscription),
+            Arg.Is(tenant),
+            Arg.Any<RetryPolicyOptions>())
+            .Returns(expectedConnection);
+
+        var command = new DatabaseAddCommand(_logger);
+        var args = command.GetCommand().Parse([
+            "--subscription", subscription,
+            "--resource-group", resourceGroup,
+            "--app-name", appName,
+            "--database-type", databaseType,
+            "--database-server", databaseServer,
+            "--database-name", databaseName,
+            "--tenant", tenant
+        ]);
+        var context = new CommandContext(_serviceProvider);
+
+        // Act
+        var response = await command.ExecuteAsync(context, args);
+
+        // Assert
+        Assert.NotNull(response);
+        Assert.Equal(200, response.Status);
+
+        await _appServiceService.Received(1).AddDatabaseAsync(
+            appName,
+            resourceGroup,
+            databaseType,
+            databaseServer,
+            databaseName,
+            Arg.Any<string>(),
+            subscription,
+            tenant,
+            Arg.Any<RetryPolicyOptions>());
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_MissingRequiredParameter_ReturnsErrorResponse()
+    {
+        // Arrange
+        var command = new DatabaseAddCommand(_logger);
+        var args = command.GetCommand().Parse([
+            "--subscription", "sub123",
+            "--resource-group", "rg1"
+            // Missing required parameters
+        ]);
+        var context = new CommandContext(_serviceProvider);
+
+        // Act
+        var response = await command.ExecuteAsync(context, args);
+
+        // Assert
+        Assert.NotNull(response);
+        Assert.NotEqual(200, response.Status);
+
+        await _appServiceService.DidNotReceive().AddDatabaseAsync(
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<RetryPolicyOptions>());
+    }
```

Comment:
Please refer to other tests that verify this to see how they use the [Theory] annotation to test multiple scenarios on a per-parameter basis.

Reply:
done